### PR TITLE
Core update 5

### PR DESCRIPTION
**Pull request by lachlan-00**

### CODE REVIEW

File: admin/access.php [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
 UI::show_header();
 
-switch ($_REQUEST['action']) {
```
Comment (Psy-Virus) [inline admin/access.php]: Aren't those changes overkill, as we already validate the incoming data hard coded by with switches?

What I can see is that there is that we slow down the code execution by adding a function that doesn't improve the security in those cases.

The filter_input function would be interesting if we had some user controlled output, like a message or a post that's saved in a database, unfiltered/unsanitized. But in these cases I don't think it's an improvement, as code is only executed if the POST or GET request matches any case. If there's no match, the default case will handle the request or if there's no default case, nothing will happen.

Comment (lachlan-00) [reply admin/access.php]: Thanks for reviewing.

I haven't compared load times between the branches yet. But I do agree that I haven't checked whether all the filters are needed.

There are xss and injection abilities in develop for pretty much every page right now.

Once I've made all the code changes in core part of my process will be on testing between core <-> develop as well as speed of those functions.

**Comment by lachlan-00** [comment]
@psy-virus I see what you mean with the switch. The PHP manual leads me to believe that switches will only execute on the case, not the variable so because the cases are hard coded strings they shouldn't be vulnerable.

https://www.php.net/manual/en/control-structures.switch.php
```
 Only when a case statement is found whose expression evaluates to a value that matches the value of the switch expression does PHP begin to execute the statements.
```

**Comment by Psy-Virus** [comment]
Precisely ^^
I believe that those automatic checks that you probably have run (or they: #1872) to find vulnerabilities are not made for those cases and create false positives.

But all in all, good job so far. It's nice to have someone who has, before all, the time and the motivation to push Ampache and is not afraid of learning something new.

:)

**Comment by lachlan-00** [comment]
Thanks for the positive thoughts! I'll update my branch and I can filter the returning issues I know will be safe.

I'm really happy to push forward with all the "boring" things that haven't really moved over the years. It's allowed me to do a few things that have been bugging me and the Ampache project itself has been really important to me personally.

I've made a lot of the changes I wanted to make in the modernization project so I'm hoping in a month I should be able to start my testing plan to move core into develop.

**Pull request merged**